clients/cobol-lsp-vscode-extension/src/services/snippetcompletion/cobolSnippets.json

```diff
@@ -1556,32 +2117,126 @@
   "OPEN I-O": {
     "prefix": "OPEN I-O",
     "body": ["OPEN I-O ${1:fileName}"],
-    "description": "OPEN I-O FILENAME"
+    "description": "OPEN I-O FILENAME."
   },
   "OPEN INPUT": {
     "prefix": "OPEN INPUT",
-    "body": ["OPEN INPUT ${1:fileName}"],
-    "description": "OPEN INPUT FILENAME"
+    "body": ["OPEN INPUT ${1:fileName}${2|., REVERSED., WITH NO REWIND.|}"],
+    "description": "OPEN INPUT initiates the process of reading files. Note that REVERSED and WITH NO REWIND are optional clauses."
   },
   "OPEN OUTPUT": {
     "prefix": "OPEN OUTPUT",
-    "body": ["OPEN OUTPUT ${1:fileName}"],
-    "description": "OPEN OUTPUT FILENAME"
+    "body": ["OPEN OUTPUT ${1:fileName}${2|., WITH NO REWIND.|}"],
+    "description": "OPEN OUTPUT initiates the process of writing files. Note that NO REWIND is optional."
+  },
+  "OPTIONAL": {
+    "prefix": "OPTIONAL",
+    "body": ["SELECT OPTIONAL ${1:filename}", "ASSIGN TO ${2:EXTFILE}"],
+    "description": "If SELECT OPTIONAL is specified in the file-control entry for this file, and the file is not available at runtime, execution of the first READ statement causes an at-end condition - and not an ABEND. Note that this is the case for QSAM and VSAM opened for INPUT, I-O, or EXTEND."
```

Comment:
"If SELECT OPTIONAL is specified in the file-control entry for this file, and the file is not available at runtime, execution of the first READ statement causes an at-end condition rather than an ABEND. This applies to QSAM and VSAM files opened for INPUT, I-O, or EXTEND."